sthgetter/sthgetter.go

```diff
@@ -68,7 +72,72 @@ func getCheckStoreSTH(ctx context.Context, lc *client.LogClient, st storage.APIC
 		glog.Infof("%s: %s: error writing API Call %s: %s", l.URL, logStr, apiCall, err)
 	}
 
-	//TODO(katjoyce): Run checks on the received STH.
+	if sth == nil {
+		return
+	}
+
+	// Verify the STH.
+	errs := checkSTH(sth, apiCall.End, sv, l)
+	if len(errs) != 0 {
+		var b strings.Builder
+		fmt.Fprintf(&b, "STH verification errors for STH %v:", sth)
+		for _, e := range errs {
+			fmt.Fprintf(&b, "\n\t%v: %v,", reflect.TypeOf(e), e)
```

Comment:
```suggestion
			fmt.Fprintf(&b, "\n\t%T: %v", e, e)
```

Reply:
Alternatively:
```suggestion
			fmt.Fprintf(&b, "\n\t%#v", e)
```

Reply:
Done the first one.  I tried the second one originally, and what it prints out is very messy, containing addresses of the inner error object and duplication etc.  Opted for just printing the type and the error message as it was a bit prettier :)